src/superphot_plus/tune_model.py

```diff
@@ -0,0 +1,256 @@
+                lc_names=train_names,
+                labels=train_classes,
+                goal_per_class=config["goal_per_class"],
+                fits_dir=FITS_DIR,
+                sampler=SAMPLER,
+                redshifts=train_redshifts,
+            )
+            val_features, val_classes, val_redshifts = oversample_using_posteriors(
+                lc_names=val_names,
+                labels=val_classes,
+                goal_per_class=round(0.1 * config["goal_per_class"]),
+                fits_dir=FITS_DIR,
+                sampler=SAMPLER,
+                redshifts=val_redshifts,
+            )
+
+            train_features = np.hstack(
+                (
+                    train_features,
+                    np.array(
+                        [
+                            train_redshifts,
+                        ]
+                    ).T,
+                )
+            )
+            val_features = np.hstack(
+                (
+                    val_features,
+                    np.array(
+                        [
+                            val_redshifts,
+                        ]
+                    ).T,
+                )
+            )
+        else:
+            train_features, train_classes = oversample_using_posteriors(
+                lc_names=train_names,
+                labels=train_classes,
+                goal_per_class=config["goal_per_class"],
+                fits_dir=FITS_DIR,
+                sampler=SAMPLER,
+            )
+            val_features, val_classes = oversample_using_posteriors(
+                lc_names=val_names,
+                labels=val_classes,
+                goal_per_class=round(0.1 * config["goal_per_class"]),
+                fits_dir=FITS_DIR,
+                sampler=SAMPLER,
+            )
+
+        # Normalize the log distributions.
+        train_features = adjust_log_dists(train_features, redshift=INCLUDE_REDSHIFT)
+        val_features = adjust_log_dists(val_features, redshift=INCLUDE_REDSHIFT)
+        train_features, mean, std = normalize_features(train_features)
+        val_features, mean, std = normalize_features(val_features, mean, std)
+
+        # Convert to Torch DataSet objects.
+        train_dataset = create_dataset(train_features, train_classes)
+        val_dataset = create_dataset(val_features, val_classes)
+
+        model = SuperphotClassifier(
+            config=ModelConfig(
+                input_dim=train_features.shape[1],
+                output_dim=output_dim,
+                neurons_per_layer=config["neurons_per_layer"],
+                num_hidden_layers=config["num_hidden_layers"],
+                batch_size=config["batch_size"],
+                learning_rate=config["learning_rate"],
+                normalization_means=mean.tolist(),
+                normalization_stddevs=std.tolist(),
+            ),
+            train_data=TrainData(train_dataset, val_dataset),
```

Comment:
If you do keep `test_data` as a class variable for `SuperphotClassifier`, I would recommend adding

`test_data=None,  # Only optimize over training and validation`

to make the usage clear to future users.